chain/pool/src/lib.rs

```diff
@@ -86,8 +90,10 @@ impl TransactionPool {
             if remove_entry {
                 self.transactions.remove(&key);
             }
-            for hash in hashes {
-                self.unique_transactions.remove(&hash);
+            for hash in &hashes {
+                if self.unique_transactions.remove(&hash) {
+                    metrics::TRANSACTION_POOL_TOTAL.dec();
```

Comment:
do we have any way of checking this value in the unittests below?

Reply:
Unittesting is complicated because metrics are global variables. As unittests run in parallel, those values get randomly overwritten.
That can be fixed by adding a layer of abstraction - a local object proxying metric operations - but that adds quiet a lot of complexity.